zeebe/qa/integration-tests/src/test/java/io/camunda/zeebe/it/processing/TimerTriggerBannedInstanceIT.java

```diff
@@ -0,0 +1,124 @@
+  }
+
+  /**
+   * Given a process with a timer event that triggers some time in the future, when the instance is
+   * banned before the timer is triggered, we should not produce many Timer TRIGGER commands for
+   * that timer.
+   *
+   * <p>This was a problem previously because every time the DueDateChecker ran, it finds the timer
+   * instance of the banned instance. Since, we remove the timer record from the command cache after
+   * writing the record to the log, we write banned instance's timer record to the command log again
+   * and again.
+   */
+  @RegressionTest("https://github.com/camunda/camunda/issues/14213")
+  public void shouldTriggerTimerEventOnlyOnceWhenInstanceIsBanned() {
+    // given
+    final var processId = "process1";
+    client
+        .newDeployResourceCommand()
+        .addProcessModel(
+            Bpmn.createExecutableProcess(processId)
+                .startEvent("start")
+                .intermediateCatchEvent("timer", b -> b.timerWithDurationExpression("duration"))
+                .endEvent("end")
+                .done(),
+            "process1.bpmn")
+        .send()
+        .join();
+
+    final long processInstanceKey =
+        client
+            .newCreateInstanceCommand()
+            .bpmnProcessId(processId)
+            .latestVersion()
+            .variables(Map.of("duration", "PT2S"))
+            .send()
+            .join()
+            .getProcessInstanceKey();
+
+    RecordingExporter.timerRecords()
+        .withIntents(TimerIntent.CREATED)
+        .withProcessInstanceKey(processInstanceKey)
+        .await();
+
+    // when
+    final BanningActuator actuator = BanningActuator.of(zeebe);
+    actuator.ban(processInstanceKey);
+
+    // create a timer instance with a longer due date, so that banned instance's timer is also
+    // triggered
+    final long firstTriggeredProcessInstanceKey =
+        client
+            .newCreateInstanceCommand()
+            .bpmnProcessId(processId)
+            .latestVersion()
+            .variable("duration", "PT3S")
+            .send()
+            .join()
+            .getProcessInstanceKey();
+
+    RecordingExporter.timerRecords(TimerIntent.TRIGGERED)
+        .withProcessInstanceKey(firstTriggeredProcessInstanceKey)
+        .await();
+
+    final long shorterDueDateProcessInstanceKey =
+        client
+            .newCreateInstanceCommand()
+            .bpmnProcessId(processId)
+            .latestVersion()
+            .variable("duration", "PT1S")
+            .send()
+            .join()
+            .getProcessInstanceKey();
+
+    RecordingExporter.timerRecords(TimerIntent.TRIGGERED)
+        .withProcessInstanceKey(shorterDueDateProcessInstanceKey)
+        .await();
+
+    // then
+    Assertions.assertThat(
+            RecordingExporter.timerRecords()
+                .withProcessInstanceKey(processInstanceKey)
+                .limit(t -> t.getIntent() == TimerIntent.TRIGGERED)
```

Comment:
❌ This limit is placed wrongly causing slow test execution. 

There is no `TRIGGERED` event on the banned process instance, only on the second (non-banned) process instance. So, this limit simply waits 5 seconds and then continues. To ensure a fast test, you'll need to flip the order of these calls. First limit for a `TRIGGERED` event, then filter the stream for the records of the banned process instance, before finally asserting that only a single `TRIGGER` command exists.